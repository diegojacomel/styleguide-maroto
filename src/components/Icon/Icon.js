// Modules
import React from "react";
import PropTypes from 'prop-types';

// Styles
import styled from "styled-components";

// Svg
import iconsList from './iconsList.svg';

const IconStyle = styled.svg`
    display: block;
    width: ${props => props.theme.spacing[props.iconSize]};
    height: ${props => props.theme.spacing[props.iconSize]};
    fill: ${props => props.theme.colors[props.iconColor]};
`;

const Icon = ({ tag, iconSize, iconColor, }) => (
    <IconStyle
        viewBox="0 0 32 32"
        className={`icon icon-${tag}`}
        iconSize={iconSize}
        iconColor={iconColor}
        data-testid="icon"
    >
        <use xlinkHref={`${iconsList}#icon-${tag}`} />
    </IconStyle>
);

Icon.propTypes = {
    tag: PropTypes.string
}

Icon.defaultProps = {
    iconSize: 'md'
}

export default Icon;