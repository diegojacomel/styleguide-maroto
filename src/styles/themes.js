// Scales
import { pxToRem } from '../utils/scales';

const theme = {
    colors: {
        primary: "#54a0ff",
        danger: "#ff6b6b",
        info: "#48dbfb",
        success: "#1dd1a1",
        default: "#576574",
        white: "#ffffff",
        black: "#000000",
        greyLight: "#f2f2f2",
        greyMedium: "#999999",
        greyDark: "#333333",
        hover: {
            primary: "#1a81ff",
            danger: "#ff3333",
            info: "#05cdfa",
            success: "#159d79",
            default: "#373f49",
            white: "#f2f2f2",
            black: "#1a1a1a"
        }
    },
    spacing: {
        none: "0",
        xxs: pxToRem(3),
        xs: pxToRem(5),
        xsm: pxToRem(8),
        sm: pxToRem(10),
        md: pxToRem(15),
        lg: pxToRem(20),
        xlg: pxToRem(25),
        xg: pxToRem(30),
        xxg: pxToRem(50)
    },
    fontSize: {
        none: "0",
        xs: pxToRem(12),
        sm: pxToRem(14),
        md: pxToRem(16),
        lg: pxToRem(18),
        xg: pxToRem(20),
    },
    rounded: {
        none: "0",
        sm: "3px",
        md: "5px",
        lg: "10px",
        full: "100%"
    }
};

export {
    theme
};