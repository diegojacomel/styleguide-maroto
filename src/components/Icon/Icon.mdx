---
name: Icon
route: /icon
---

import Icon from './Icon'
import { Playground } from 'docz'
import { ThemeProvider } from 'styled-components'
import { theme } from '../../styles/themes';
import { arrayIcons } from '../../utils/icons';

# Icon

## Basic usage
<Playground>
    <ThemeProvider theme={theme}>
        <div style={{ display: 'flex', alignItems: 'center', flexWrap: 'wrap' }}>
            {
                arrayIcons.map((icon, index) => {
                    return (
                        <div key={index} style={{ flexBasis: '25%', display: 'flex', alignItems: 'center' }}>
                            <Icon tag={icon} />
                            <p style={{ marginLeft: '15px' }}>
                                icon-{icon}
                            </p>
                        </div>
                    )
                })
            }
        </div>
    </ThemeProvider>
</Playground>