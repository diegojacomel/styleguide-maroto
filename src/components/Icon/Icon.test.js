// Modules
import React from 'react';
import { render, cleanup } from 'react-testing-library';
import { ThemeProvider } from 'styled-components';
import { theme } from '../../styles/themes';

// Components
import Icon from './Icon';

// Unmount and cleanup DOM after the test is finished
afterEach(cleanup)

describe("Test Icon component", () => {
    // Tests

    test("It's present in the document", () => {
        const { getByTestId, } = render(
            <ThemeProvider theme={theme}>
                <Icon />
            </ThemeProvider>
        );
        
        const element = getByTestId('icon');

        expect(element).toBeInTheDocument();
    })

    test("Generate snapshots", () => {
        const { container, } = render(
            <ThemeProvider theme={theme}>
                <Icon tag="home" iconSize="md" iconColor="white" />
            </ThemeProvider>
        );

        expect(container.firstChild).toMatchSnapshot();
    })
})