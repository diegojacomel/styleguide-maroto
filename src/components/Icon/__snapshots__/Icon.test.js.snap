// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Icon component Generate snapshots 1`] = `
<svg
  class="icon icon-home sc-bdVaJa dwmXVo"
  data-testid="icon"
  viewBox="0 0 32 32"
>
  <use
    xlink:href="iconsList.svg#icon-home"
  />
</svg>
`;
